### PR TITLE
fixing the DSC node registration command

### PR DESCRIPTION
**Pull request by avkaur**
## Description

Fixes: https://github.com/Azure/azure-powershell/issues/7323
fixing the DSC node registration command

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MiYanni** [comment]
@avkaur That Travis failure was fixed in our preview branch. I can pull in the latest preview into AzureRM.Automation, but it will likely cause you to have merge conflicts. Let me know.

**Comment by cormacpayne** [comment]
@avkaur ping on the above comment

**Comment by MiYanni** [comment]
@avkaur For that specific situation, do this change in your branch:
https://github.com/Azure/azure-powershell/pull/7490/files

For your other build issue, you didn't update your `psd1` files to include the new dependency. However, as mentioned already in the email chain, you can do this change for a preview of the module, but you won't be able to do this in the actual codebase. We no longer support taking new dependencies on other services' SDKs.

File: src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs [hunk 1]
```diff
@@ -790,50 +795,138 @@ public void DeleteDscNode(string resourceGroupName, string automationAccountName
...
-            templateParameters.Add("configurationMode", configurationMode);
-            templateParameters.Add("configurationModeFrequencyMins", configurationModeFrequencyMins);
-            templateParameters.Add("refreshFrequencyMins", refreshFrequencyMins);
-            templateParameters.Add("rebootNodeIfNeeded", rebootFlag);
-            templateParameters.Add("actionAfterReboot", actionAfterReboot);
-            templateParameters.Add("allowModuleOverwrite", moduleOverwriteFlag);
-            templateParameters.Add("timestamp", DateTimeOffset.UtcNow.ToString("o"));
-
             // invoke the New-AzureRmResourceGroupDeployment cmdlet
             using (var pipe = System.Management.Automation.PowerShell.Create(RunspaceMode.NewRunspace))
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs]: Please don't create a PowerShell runspace, as it is not used with your new changes.

File: src/ResourceManager/Automation/Commands.Automation/Commands.Automation.Netcore.csproj [hunk 2]
```diff
@@ -29,6 +29,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.Automation" Version="3.4.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="2.0.0-preview" />
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/Commands.Automation.Netcore.csproj]: Keep in mind you won't be able to use this when merging back to our main branch.

File: src/ResourceManager/Automation/Commands.Automation/Commands.Automation.csproj [hunk 3]
```diff
@@ -255,6 +255,10 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Automation.3.4.0-preview\lib\net452\Microsoft.Azure.Management.Automation.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Management.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.ResourceManager.2.0.0-preview\lib\net452\Microsoft.Azure.Management.ResourceManager.dll</HintPath>
+    </Reference>
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/Commands.Automation.csproj]: Keep in mind you won't be able to use this when merging back to our main branch.

File: src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs [hunk 4]
```diff
@@ -34,6 +37,7 @@
 using Job = Microsoft.Azure.Management.Automation.Models.Job;
 using JobSchedule = Microsoft.Azure.Management.Automation.Models.JobSchedule;
 using Schedule = Microsoft.Azure.Commands.Automation.Model.Schedule;
+using Microsoft.Azure.Management.ResourceManager;
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs]: Keep in mind you won't be able to use this when merging back to our main branch.

File: src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs [hunk 5]
```diff
@@ -1718,4 +1811,52 @@ private string GetNodeConfigurationListFilterString(string configurationName)
...
+        public string Schema { get; set; }
+
+        public string ContentVersion { get; set; }
+
+        public ParametersObj Parameters { get; set; }
+    }
+
+    internal class ParametersObj
+    {
+        public TemplateParameters vmName { get; set; }
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs]: Property names should not start with lower-case letters. C# uses Pascal case for properties.

File: src/ResourceManager/Automation/Commands.Automation/packages.config [hunk 6]
```diff
@@ -1,4 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Azure.Management.Automation" version="3.4.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.ResourceManager" version="2.0.0-preview" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/packages.config]: Keep in mind you won't be able to use this when merging back to our main branch.

File: src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs (hunk 4 above)
Comment (avkaur) [reply src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs]: sure, my understanding is that in the main branch all would be on latest version. as we saw that Profile module is dependent on the older version of resources module

**Comment by avkaur** [comment]
updated the review comments.
for the test didn't make any update, as test doesn't need any dependency on this new reference

File: src/ResourceManager/Automation/Commands.Automation/AzureRM.Automation.psd1 [hunk 7]
```diff
@@ -55,6 +55,7 @@ RequiredModules = @(@{ModuleName = 'AzureRM.Profile'; ModuleVersion = '5.5.1'; }
 
 # Assemblies that must be loaded prior to importing this module
 RequiredAssemblies = '.\Microsoft.Azure.Management.Automation.dll'
+RequiredAssemblies = '.\Microsoft.Azure.Management.ResourceManager.dll'
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation/AzureRM.Automation.psd1]: This is incorrect. It should be:
```powershell
RequiredAssemblies = '.\Microsoft.Azure.Management.Automation.dll',
    '.\Microsoft.Azure.Management.ResourceManager.dll'
```

File: src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs (hunk 4 above)
Comment (MiYanni) [reply src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClientDSC.cs]: Modules on the main branch use our [internal version of the ResourceManager SDK](https://github.com/Azure/azure-powershell-common/tree/develop/src/ResourceManager). A new version of this SDK would need to be created in our Common repo. Then, your changes would need to use that version. This is not necessary for a preview module, but will not be allowed in our main repo.

**Comment by MiYanni** [comment]
@avkaur Ping

**Comment by avkaur** [comment]
I have made changes as needed and the static analysis fix. lets see the build now

**Pull request merged**